### PR TITLE
feat(corrections): show original scan alongside extracted text (#1314)

### PR DESCRIPTION
**Pull request by Robert-Freire**
## Summary

- When a teacher uploads a scanned JPG/PNG/WEBP/PDF, the CorrectionDrawer expands from 520px to 90vw and shows the original document on the left with the editable OCR text on the right
- The scan appears instantly via `URL.createObjectURL` before the OCR API call returns, so the teacher has a visual reference during extraction
- PDF files render in a native `<iframe>` (browser handles multi-page scroll + zoom); images render with `` + `object-fit: contain`
- DOCX uploads and pasted text stay in the single-column narrow layout (no scan to show)
- On OCR error the scan stays visible so the teacher can still type manually while comparing
- HEIC or DOCX uploaded after a valid image clears the stale preview
- Object URLs are revoked on drawer close/unmount (no memory leak)

## Design decisions (resolved with Vera before implementation)

- Progressive expansion (narrow → wide on file select) avoids a jarring layout jump for DOCX/paste flows
- `surface-container-low` (#F4F2FD) left panel with no divider line (compliant with Academic Atelier no-line rule)
- No custom zoom controls: PDFs in iframe get browser zoom natively; images use `object-fit: contain` with scroll

## Test plan

- [ ] Build verify: PASS (bicep, dotnet build, dotnet test, npm lint, npm build, npm test 110/110)
- [ ] Unit tests: 35 pass (6 new tests: img/PDF/DOCX/error/wide-class/revoke)
- [ ] Verify: unit tests cover all behaviors; full Playwright e2e blocked by CORS (dev server vs e2e API) and pre-change container -- DOCX behavior confirmed on e2e stack
- [ ] qa-verify: ran (found pre-existing issue with wrong issue scope; unit tests confirm all AC)
- [ ] Code review: PASS WITH NOTES -- two stale-preview edge cases (HEIC-after-image, DOCX-after-image) fixed and covered by tests
- [ ] review-ui: ALMOST -- two-column expanded state not visually confirmed (e2e container pre-change); narrow drawer and empty state comply with Academic Atelier DS

## Out-of-scope observations

- Pre-existing: CorrectionDrawer footer Ghost+Primary pairing violates DS §5 (logged to observed-issues)
- Pre-existing: Manrope Display-LG font not rendering in e2e stack for Redacciones empty state (logged to observed-issues)
- New pattern: `surface-container-low` as image viewer panel bg flagged to ui-review-backlog for Vera to document

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added local file preview in the correction drawer—images and PDFs display immediately upon upload, while DOCX files skip preview.
  * Preview remains visible if OCR extraction fails, allowing manual editing.
  * Drawer layout adjusts to accommodate preview alongside form fields.
  * Preview properly clears when uploading a different file or closing the drawer.

[Review Change Stack](https://app.coderabbit.ai/change-stack/Robert-Freire/langTeachSaaS/pull/1346?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: frontend/src/components/student/RedaccionesTab.tsx [hunk 1]
```diff
@@ -330,14 +332,36 @@ function CorrectionDrawer({ studentId, editId, onClose, onSaved, onCorregirError
...
+
   async function handleFileUpload(file: File) {
     setOcrErrorCode(null)
     const lower = file.name.toLowerCase()
     if (HEIC_EXTENSIONS.some((ext) => lower.endsWith(ext))) {
+      setLocalPreviewUrl(null)
+      setPreviewType(null)
       setOcrError('El formato HEIC no es compatible. Usa JPG, PNG, WEBP o PDF.')
       setOcrState('error')
       return
```
[the remark this answers is not in the document]
Comment (Robert-Freire) [reply frontend/src/components/student/RedaccionesTab.tsx:351]: Fixed in 0808a673 — added `setBlobUrl(null)` to the HEIC rejection branch so a prior OCR's blob URL cannot leak into `sourceImageUrl` on save.
